Divine/CLI/CommandLineLogger.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using Divine.Enums;
+
+namespace Divine.CLI
+{
+    internal class CommandLineLogger
+    {
+        private static readonly LogLevel OptionLogLevel = CommandLineActions.LogLevel;
+
+        public static void LogFatal(string message, int errorCode) => Log(LogLevel.FATAL, message, errorCode);
+        public static void LogError(string message) => Log(LogLevel.ERROR, message);
+        public static void LogWarn(string message) => Log(LogLevel.WARN, message);
+        public static void LogInfo(string message) => Log(LogLevel.INFO, message);
+        public static void LogDebug(string message) => Log(LogLevel.DEBUG, message);
+        public static void LogTrace(string message) => Log(LogLevel.TRACE, message);
+        public static void LogAll(string message) => Log(LogLevel.ALL, message);
+
+        private static void Log(LogLevel loglevel, string message, int errorCode = -1)
+        {
+            if (OptionLogLevel == LogLevel.OFF)
+            {
+                return;
+            }
+
+            switch (loglevel)
+            {
+                case LogLevel.FATAL:
+                    if (OptionLogLevel > LogLevel.OFF)
+                    {
+                        Console.WriteLine($"[FATAL] {message}");
+                    }
+
+                    if (errorCode == -1)
+                    {
+                        Environment.Exit((int) LogLevel.FATAL);
+                    }
+                    else
+                    {
+                        Environment.Exit((int) LogLevel.FATAL + errorCode);
+                    }
+                    break;
+
+                case LogLevel.ERROR:
+                    if (OptionLogLevel < LogLevel.ERROR)

```

Comment:
Wouldn't it be easier to just check `if (OptionLogLevel < loglevel) { ... }` before the switch?

Reply:
LOL. Serves me right for working late in multiple languages.